wire-schema/src/main/java/com/squareup/wire/schema/Reserved.java

```diff
@@ -13,74 +13,29 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.squareup.wire.schema;
+package com.squareup.wire.schema
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Range;
-import com.squareup.wire.schema.internal.parser.ReservedElement;
-import java.util.List;
+import com.google.common.collect.Range
+import com.squareup.wire.schema.internal.parser.ReservedElement
 
-final class Reserved {
-  private final Location location;
-  private final String documentation;
-  private final List<Object> values;
-
-  private Reserved(Location location, String documentation, List<Object> values) {
-    this.location = location;
-    this.documentation = documentation;
-    this.values = values;
-  }
-
-  public Location location() {
-    return location;
+class Reserved(
+  val location: Location,
+  val documentation: String,
+  val values: List<Any>
+) {
+  fun matchesTag(tag: Int) = values.any {
+    it is Int && tag == it || (it as? Range<Int>)?.contains(tag) == true
```

Comment:
I wonder if `tag in (it as? Range<Int>)` will compile without safe-calls

Reply:
<img width="936" alt="Screen Shot 2019-04-01 at 11 45 12 PM" src="https://user-images.githubusercontent.com/1868149/55374946-4fccc600-54d8-11e9-9fcd-bfaaadbed09d.png">

Nope.